### PR TITLE
[Binding] enable mlir python binding for graph compiler

### PR DESCRIPTION
**Pull request by xurui1995**
Enable basic mlir python binding for `onednngraph dialect`  and passes in `gc/Transforms`, will add support for more dialects and related passes in the next PR.
Tracking https://github.com/intel/graph-compiler/issues/57

### CODE REVIEW

**Comment by xurui1995** [comment]
@WangJialei-A you can develop bench gc based on bindings of this PR

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
+include(AddMLIRPython)
+
+# Specifies that all MLIR packages are co-located under the `gc_mlir`
+# top level package (the API has been embedded in a relocatable way).
+# TODO: Add an upstream cmake param for this vs having a global here.
+add_compile_definitions("MLIR_PYTHON_PACKAGE_PREFIX=gc_mlir.")
+
+
```
Comment (Menooker) [inline python/CMakeLists.txt:8]: can we add a cmake option to turn off the pybind even if LLVM has python enabled?

File: python/MainModule.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
+/*
+ * Copyright (C) 2024 Intel Corporation
```
Comment (Menooker) [inline python/MainModule.cpp:3]: Please use LLVM license style here. Reference: https://github.com/intel/graph-compiler/pull/85

Comment (WangJialei-A) [reply python/MainModule.cpp:3]: #85 is merged. Please rebase this PR and have a license check

File: python/CMakeLists.txt (hunk 1 above)
Comment (xurui1995) [reply python/CMakeLists.txt:8]: > can we add a cmake option to turn off the pybind even if LLVM has python enabled?

yes， I can add a new  cmake option for the gc pybind, should we set the default value to OFF?

Comment (Menooker) [reply python/CMakeLists.txt:8]: @ZhennanQin Do we need pybind by default? Or should we only target onednn graph api for now?

Comment (ZhennanQin) [reply python/CMakeLists.txt:8]: Suggest to enable pybind by default as this is the way for release. For develop purposes, you can turn it off with extra cmake flag.

Comment (xurui1995) [reply python/CMakeLists.txt:25]: option added in https://github.com/intel/graph-compiler/blob/dda942cb46a0fffb658a3b360ab31fff970e7cd0/CMakeLists.txt#L54C1-L57C8

File: CMakeLists.txt [hunk 3]
```diff
@@ -33,6 +51,16 @@ include(AddLLVM)
 include(AddMLIR)
 include(HandleLLVMOptions)
 
+if(GC_ENABLE_BINDINGS_PYTHON AND NOT MLIR_ENABLE_BINDINGS_PYTHON)
+  message(STATUS "Enabling Python API failed due to the 'MLIR_ENABLE_BINDINGS_PYTHON' for LLVM is not ON.")
```
Comment (Menooker) [inline CMakeLists.txt:55]: Wording: Failed to enable Python API ...

Comment (Menooker) [reply CMakeLists.txt]: Would you please also update the readme of the for the cmake command. The main branch has updated the readme now.

Comment (xurui1995) [reply CMakeLists.txt]: fixed

Comment (xurui1995) [reply CMakeLists.txt]: > Would you please also update the readme of the for the cmake command. The main branch has updated the readme now.

have updated the readme of main project, and I will update a more specific readme in the python dir after binding more dialects in the future

**Comment by xurui1995** [comment]
@WangJialei-A @Menooker could you please help approve this PR if it looks good to you, thanks!

**Comment by WangJialei-A** [comment]
Hi, @xurui1995 
Please place an empty `__init__.py` file under folder `gc_mlir`, `gc_mlir/dialects` in order to import it as a normal module.

**Review by WangJialei-A — APPROVED** [review]

**Pull request merged**